### ISSUE TITLE
check cursor+tab boolean only once

### ISSUE THREAD

**Issue by esaruoho** [surge-synthesizer/surge#729]
"This is ok for now but we should clean it up to do that check once and set a bool. If you push this and make a new ticket I can do that or you can" from @baconpaul 

i tried to make it like this:
<img width="677" alt="image" src="https://user-images.githubusercontent.com/4966687/53813429-4e09f400-3f66-11e9-8029-f54d7a2e4b9d.png">

not sure if that's valid though

**Comment by esaruoho** [comment]
yeah that picture broke everything. oops.
well, looks like i'll have to learn something by seeing how it's actually done.

**Comment by baconpaul** [comment]
I would put it above even the first if

**Comment by baconpaul** [comment]
So

```
int32_t SurgeGUIEditor::onKeyDown(const VstKeyCode& code, CFrame* frame)
{
   if (saveDialog && saveDialog->isVisible())
   {
      /* 
      ** SaveDialog gets access to the tab key to switch between fields if it is open
      */
      return -1;
   }
   
    if(code.virt != 0 )
    {
```

**Comment by baconpaul** [comment]
and modify the comment to say 'if the save dialog is showing let it have all the keys'

But I think we may also want to kill the arrows altogether. Too dangerous. You hit one mid patch edit and you lose your patch. I'm fine for them to all go.

(What we really want is a chicken box if you are changing patches and have unsaved changes but that is very hard. If you decide to add an issue for that we should link the issue to undo/redo also)

**Comment by esaruoho** [comment]
yeah.. i was not a fan of cursor up / down / left / right for changing patches. trackers, for instance, live and breathe on updownleftright, and since vst/au are floaty windows on top of that, the chances of you trying to add a change in a channel on the tracker editor by pressing a cursor key, while having the au/vst plugin window open, and losing a preset is pretty high.

**Comment by baconpaul** [comment]
Yeah so OK I would just remove them. Put the saveDialog at the top. And then tag the other key issues as 1.6.n

**Comment by esaruoho** [comment]
#733 added for warning dialog

**Comment by esaruoho** [comment]
remove cursor keys issue added #734
